src/index.ts

```diff
@@ -39,10 +43,35 @@ class MagicNixCacheAction {
         ],
       },
     });
+
+    if (process.env[ENV_CACHE_DAEMONDIR]) {
+      this.unsafeDaemonDir = process.env[ENV_CACHE_DAEMONDIR];
+    } else {
+      this.unsafeDaemonDir = this.idslib.getTemporaryName();
+      actionsCore.exportVariable(ENV_CACHE_DAEMONDIR, this.unsafeDaemonDir);
+    }
+    this.idslib.stapleFile(
+      "daemon.log",
+      path.join(this.unsafeDaemonDir, "daemon.log"),
+    );
+  }
+
+  async getDaemonDir(): Promise<string> {
+    if (this.daemonDir === undefined) {
+      await mkdir(this.unsafeDaemonDir, { recursive: true });
```

Comment:
If we make this the synchronous version (if it exists..................), we could skip having an `unsafeDaemonDir` altogether by just... creating it, right?